### PR TITLE
Move CI to GHA

### PR DESCRIPTION
**Pull request by Molter73**
## Description

This is the final step for making GHA the official CI solution, OSCI needs to be taken offline first. The PR will remain in draft in order to prevent OSCI from running (GHA jobs trigger regardless of this), once OSCI is disabled we can open and merge this PR directly.

## Checklist
- [x] Investigated and inspected CI test results

## Testing Performed

Testing as much of the affected CI jobs as possible through the `build-builder-image` and `no-cache` labels, as well as testing the support-packages. final test will need to happen on the master branch though.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -98,10 +98,12 @@ jobs:
       branch-name: ${{ needs.build-drivers.outputs.branch-name }}
       drivers-bucket: ${{ needs.init.outputs.drivers-bucket }}
       support-packages-bucket: ${{ needs.init.outputs.support-packages-bucket }}
+      public-support-packages-bucket: ${{ needs.init.outputs.public-support-packages-bucket }}
     # If you want to test this workflow in a PR, you will need to replace
     # this condition with `${{ always() }}`, since skipping the upload of
     # bundles will cause this workflow to skip as well.
-    if: ${{ github.event_name != 'pull_request' }}
+    # if: ${{ github.event_name != 'pull_request' }}
+    if: always()
```
Comment (Stringy) [inline .github/workflows/main.yml:106]: just a note to revert this before merging. 

Also would it be worth adding a label to trigger this behaviour?

**Review by Stringy — APPROVED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (Molter73) [reply .github/workflows/main.yml:106]: Maybe we could achieve this with a condition similar to this:
```yml
if: always() && (github.event_name == 'push' || contains(github.event.pull_request.labels.*.name, 'test-support-package')
```
I think it is a little hard to read, WDYT?

Comment (Stringy) [reply .github/workflows/main.yml:106]: yeah I agree it's pretty complex as a condition but I think it'll make testing a lot easier and won't risk someone merging `if: always()`

Comment (Molter73) [reply .github/workflows/main.yml:106]: I'll add it and get it tested 🙂

Comment (Molter73) [reply .github/workflows/main.yml]: Looks like it is working properly.

**Review by Stringy — APPROVED** [review]

**Pull request merged**